app/browser/tabs.js

```diff
@@ -473,10 +474,35 @@ const api = {
       tab.on('unresponsive', () => {
         console.log('unresponsive')
       })
+
       tab.on('responsive', () => {
         console.log('responsive')
       })
 
+      tab.on('tab-changed-at', () => {
+        updateTab(tabId)
+      })
+
+      tab.on('tab-moved', () => {
+        appActions.tabMoved(tabId)
+      })
+
+      tab.on('will-attach', () => {
+        appActions.tabWillAttach(tab.getId())
+      })
+
+      tab.on('tab-strip-empty', () => {
+        tab.once('will-attach', () => {
+          const tabValue = getTabValue(tabId)
```

Comment:
couldn't this happen in `appActions.tabWillAttach` instead of having a second action for `tabStripEmpty`?